### PR TITLE
Fix: use sourceKey to extract value from ConfigMap/Secret

### LINKED ISSUE #838: Service binding is creating a secret with wrong key when sourceKey is mentioned in annotation
I was trying to bind a backend service to an application using annotation in CRD
```
service.binding/certificate: path={.status.data.dbConfiguration},objectType=ConfigMap,sourceKey=certificate
```

This is CR:
```
apiVersion: stable.example.com/v1
kind: Backend
metadata:
    name: cmsa-1-service
spec:
    image: docker.io/postgres
    imageName: postgres
    dbName: db-demo
status:
    data:
        dbConfiguration: cmsa-1-configmap    # ConfigMap
```
This is CM:
```
apiVersion: v1
kind: ConfigMap
metadata:
    name: cmsa-1-configmap
data:
    certificate: "certificate value"
```
This is sb cr:
```
apiVersion: operators.coreos.com/v1alpha1
kind: ServiceBinding
metadata:
    name: cmsa-1
spec:
    services:
      - group: stable.example.com
        version: v1
        kind: Backend
        name: cmsa-1-service
    application:
        name: cmsa-1
        group: apps
        version: v1
        resource: deployments
```

Expected intermediate secret data:

```
BACKEND_CERTIFICATE: Y2VydGlmaWNhdGUgdmFsdWU=
```

Actual intermediate secret data:
```
BACKEND: Y2VydGlmaWNhdGUgdmFsdWU=
```

Acceptance test to reproduce: https://github.com/redhat-developer/service-binding-operator/pull/842

### PR DESCRIPTION
**Pull request by akashshinde**
Fixes: #838

### CODE REVIEW

**Review by pedjak — CHANGES_REQUESTED** [review]
please incorporate into this PR the test from https://github.com/redhat-developer/service-binding-operator/pull/842 and enable it, so that we can be sure that the bug has been fixed.

Please craft a proper commit message explain what was really done, also create a proper PR description.

File: pkg/binding/model.go [hunk 1]
```diff
@@ -95,16 +95,24 @@ func newModel(annotationValue string) (*model, error) {
...
 
+	sourceValue := ""
+	sourceValue, found = raw[sourceValueModelKey]
+	if !found {
+		sourceValue, found = raw[sourceKeyModelKey]
+		if !found {
+			sourceValue = ""
+		}
+	}
+
```
[the remark this answers is not in the document]
Comment (akashshinde) [reply pkg/binding/model.go:106]: `sourceValue` is being used to figure out key fields from `ConfigMap`/`Secret`, and there few use cases where we require `sourceValue` along with `sourceKey`, ref: https://github.com/redhat-developer/service-binding-operator/blob/4f7892420a1382e3ea0ea2b0f0d2e08c8073d805/pkg/binding/spec_test.go#L253

File: pkg/binding/model.go [hunk 2]
```diff
@@ -95,16 +95,24 @@ func newModel(annotationValue string) (*model, error) {
 		sourceKey = ""
 	}
 
+	sourceValue := ""
```
[the remark this answers is not in the document]
Comment (akashshinde) [reply pkg/binding/model.go:98]: not required, removed

**Comment by wtrocki** [comment]
@akashshinde Amazing PR! Thank you so much for fix

File: test/acceptance/features/bindAppToServiceUsingConfigMap.feature [hunk 3]
```diff
@@ -1,3 +1,4 @@
+@disabled-sourcekey
```
Comment (akashshinde) [inline test/acceptance/features/bindAppToServiceUsingConfigMap.feature:1]: Need to remove this annotation

**Comment by pedjak** [comment]
@wtrocki would you like to try out this PR and verify if the issue is solved before we merge it?

**Comment by wtrocki** [comment]
This fixes problems with reflecting sourceKey but still have problem with binding 

```

    // This works after applying this PR but wasn't work before
    service.binding/password: >-
      path={.status.serviceAccountSecretName},objectType=Secret,sourceKey=client-secret

   // This doesn't work
   service.binding/bootstrapServer: 'path={.status.bootstrapServer.host}'

```

The second problem might be something to do with my setup as I see tests on operator doing the same.

**Review by wtrocki — APPROVED** [review]

**Pull request merged**